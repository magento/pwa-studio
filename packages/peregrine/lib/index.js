import * as RestApi from './RestApi';
import * as Util from './util';

// hooks
export { useEventListener } from './hooks/useEventListener';
export { useCarousel } from './hooks/useCarousel';
export { useDropdown } from './hooks/useDropdown';
export { usePagination } from './hooks/usePagination';
export { useRestApi } from './hooks/useRestApi';
export { useRestResponse } from './hooks/useRestResponse';
export { useScrollLock } from './hooks/useScrollLock';
export { useSearchParam } from './hooks/useSearchParam';
export { useSort } from './hooks/useSort';
export { useTypePolicies } from './hooks/useTypePolicies';

export {
    WindowSizeContextProvider,
    useWindowSize
} from './hooks/useWindowSize';
export { getToastId, useToasts, ToastContextProvider } from './Toasts';

// store
export { enhancer, reducers } from './store';

// components
export { default as ContainerChild } from './ContainerChild';
export { default as List, Items, Item } from './List';
export { default as Price } from './Price';
export { default as Router } from './Router';
export {
    default as PeregrineContextProvider
} from './PeregrineContextProvider';

// misc
export { RestApi };
export { Util };
export { default as createTestInstance } from './util/createTestInstance';
