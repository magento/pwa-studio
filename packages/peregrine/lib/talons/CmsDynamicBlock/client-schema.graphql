# Save cart conditions data in cache
extend type DynamicBlocks {
    salesRulesData: String
}
