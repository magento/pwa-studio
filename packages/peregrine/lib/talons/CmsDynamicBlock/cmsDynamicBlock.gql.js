import { gql } from '@apollo/client';

export const GET_CMS_DYNAMIC_BLOCKS = gql`
    query GetCmsDynamicBlocks(
        $cartId: String
        $productId: ID
        $type: DynamicBlockTypeEnum!
        $locations: [DynamicBlockLocationEnum]
        $uids: [ID]!
    ) {
        dynamicBlocks(
            input: {
                cart_id: $cartId
                product_uid: $productId
                dynamic_block_uids: $uids
                locations: $locations
                type: $type
            }
        ) {
            items {
                content {
                    html
                }
                uid
            }
            salesRulesData @client
        }
    }
`;

export const GET_SALES_RULES_DATA = gql`
    query SalesRulesDataQuery($cartId: String!) {
        cart(cart_id: $cartId) {
            id
            # eslint-disable-next-line @graphql-eslint/require-id-when-available
            items {
                uid
                # eslint-disable-next-line @graphql-eslint/require-id-when-available
                product {
                    uid
                    ... on PhysicalProductInterface {
                        weight
                    }
                }
                quantity
                # eslint-disable-next-line @graphql-eslint/require-id-when-available
                ... on ConfigurableCartItem {
                    # eslint-disable-next-line @graphql-eslint/require-id-when-available
                    configured_variant {
                        uid
                        ... on PhysicalProductInterface {
                            weight
                        }
                    }
                }
            }
            prices {
                subtotal_excluding_tax {
                    value
                }
                subtotal_including_tax {
                    value
                }
            }
            selected_payment_method {
                code
            }
            shipping_addresses {
                country {
                    code
                }
                postcode
                region {
                    code
                    region_id
                }
                street
                selected_shipping_method {
                    method_code
                }
            }
            total_quantity
        }
    }
`;

export const GET_STORE_CONFIG_DATA = gql`
    query getStoreConfigData {
        # eslint-disable-next-line @graphql-eslint/require-id-when-available
        storeConfig {
            product_url_suffix
        }
    }
`;

export const GET_PRODUCT_DETAIL_QUERY = gql`
    query getProductDetailForProductPage($urlKey: String!) {
        products(filter: { url_key: { eq: $urlKey } }) {
            # eslint-disable-next-line @graphql-eslint/require-id-when-available
            items {
                uid
                url_key
            }
        }
    }
`;

export default {
    getCmsDynamicBlocksQuery: GET_CMS_DYNAMIC_BLOCKS,
    getSalesRulesDataQuery: GET_SALES_RULES_DATA,
    getStoreConfigData: GET_STORE_CONFIG_DATA,
    getProductDetailQuery: GET_PRODUCT_DETAIL_QUERY
};
