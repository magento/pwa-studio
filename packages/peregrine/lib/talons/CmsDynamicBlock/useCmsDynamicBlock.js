import { useEffect, useCallback } from 'react';
import { useQuery } from '@apollo/client';
import { useLocation } from 'react-router-dom';

import { useCartContext } from '@magento/peregrine/lib/context/cart';
import mergeOperations from '@magento/peregrine/lib/util/shallowMerge';

import DEFAULT_OPERATIONS from './cmsDynamicBlock.gql';

export const flatten = cartData => {
    const cartItems = cartData?.cart?.items || [];
    const totalWeight = cartItems.reduce((prevItem, currentItem) => {
        const { product, quantity, configured_variant } = currentItem;

        const currentWeight = configured_variant
            ? configured_variant.weight
            : product.weight || 0;

        return prevItem + currentWeight * quantity;
    }, 0);
    const shippingAddresses = cartData?.cart?.shipping_addresses || [];
    const subtotalExcludingTax =
        cartData?.cart?.prices?.subtotal_excluding_tax?.value || 0;
    const subtotalIncludingTax =
        cartData?.cart?.prices?.subtotal_including_tax?.value || 0;
    const selectedPaymentMethod =
        cartData?.cart?.selected_payment_method?.code || null;
    const shippingCountryCode = shippingAddresses[0]?.country?.code || null;
    const shippingPostCode = shippingAddresses[0]?.postcode || null;
    const shippingRegionCode = shippingAddresses[0]?.region?.code || null;
    const shippingRegionId = shippingAddresses[0]?.region?.region_id || null;
    const selectedShippingMethod =
        shippingAddresses[0]?.selected_shipping_method?.method_code || null;
    const totalQuantity = cartData?.cart?.total_quantity || 0;

    return JSON.stringify([
        totalWeight,
        subtotalExcludingTax,
        subtotalIncludingTax,
        selectedPaymentMethod,
        shippingCountryCode,
        shippingPostCode,
        shippingRegionCode,
        shippingRegionId,
        selectedShippingMethod,
        totalQuantity
    ]);
};

/**
 * This talon contains the logic for a cms dynamic block component.
 * It performs effects and returns a data object containing values for rendering the component.
 *
 * This talon performs the following effects:
 *
 * - Get cms dynamic block based on type, locations and given uids
 * - Get sales rules data from cart to refetch dynamic block query when conditions change
 * - Update the {@link CmsDynamicBlockTalonProps} values with the data returned by the query
 *
 * @function
 *
 * @param {Array} props.locations - array of locations of cms dynamic blocks
 * @param {(String|String[])} props.uids - single or multiple uids of cms dynamic blocks
 * @param {String} props.type - type of cms dynamic blocks
 * @param {CmsDynamicBlockOperations} [props.operations]
 *
 * @returns {CmsDynamicBlockTalonProps}
 *
 * @example <caption>Importing into your project</caption>
 * import { useCmsDynamicBlock } from '@magento/peregrine/lib/talons/CmsDynamicBlock/useCmsDynamicBlock';
 */
export const useCmsDynamicBlock = props => {
    const { locations, uids, type } = props;
    const operations = mergeOperations(DEFAULT_OPERATIONS, props.operations);
    const {
        getCmsDynamicBlocksQuery,
        getSalesRulesDataQuery,
        getStoreConfigData,
        getProductDetailQuery
    } = operations;

    const [{ cartId }] = useCartContext();
    const { pathname } = useLocation();

    // Get Product Data from cache
    const { data: storeConfigData, loading: storeConfigLoading } = useQuery(
        getStoreConfigData
    );
    const slug = pathname.split('/').pop();
    const productUrlSuffix = storeConfigData?.storeConfig?.product_url_suffix;
    const urlKey = productUrlSuffix ? slug.replace(productUrlSuffix, '') : slug;
    const { data: productData, loading: productDataLoading } = useQuery(
        getProductDetailQuery,
        {
            skip: !storeConfigData,
            variables: {
                urlKey
            }
        }
    );

    // @TODO: Update with uid when done in Product Root Component
    const products =
        productData?.products?.items && productData.products.items.length > 0
            ? productData.products.items
            : [];
    const productUid = products.find(item => item.url_key === urlKey)?.uid;

    const { client, loading, error, data, refetch } = useQuery(
        getCmsDynamicBlocksQuery,
        {
            variables: {
                cartId,
                type,
                locations,
                uids,
                ...(productUid ? { productId: productUid } : {})
            },
            skip: !cartId
        }
    );

    const { loading: cartLoading, data: cartData } = useQuery(
        getSalesRulesDataQuery,
        {
            variables: { cartId },
            skip: !cartId
        }
    );

    const currentSalesRulesData = flatten(cartData);
    const isLoading =
        loading || cartLoading || storeConfigLoading || productDataLoading;
    const cachedSalesRulesData = data?.dynamicBlocks?.salesRulesData;

    const updateSalesRulesData = useCallback(
        (currentData, currentSalesRulesData) => {
            client.writeQuery({
                query: getCmsDynamicBlocksQuery,
                data: {
                    dynamicBlocks: {
                        ...currentData,
                        salesRulesData: currentSalesRulesData
                    }
                },
                variables: {
                    cartId,
                    type,
                    locations,
                    uids,
                    ...(productUid ? { productId: productUid } : {})
                },
                skip: !cartId
            });
        },
        [
            cartId,
            client,
            getCmsDynamicBlocksQuery,
            locations,
            productUid,
            type,
            uids
        ]
    );

    useEffect(() => {
        if (data && cachedSalesRulesData !== currentSalesRulesData) {
            if (!cachedSalesRulesData) {
                // Save cart conditions data in cache
                updateSalesRulesData(data.dynamicBlocks, currentSalesRulesData);
            } else {
                // Refetch cms data if there's a mismatch
                refetch();
            }
        }
    }, [
        cachedSalesRulesData,
        currentSalesRulesData,
        data,
        refetch,
        updateSalesRulesData
    ]);

    return {
        loading: isLoading,
        error,
        data
    };
};

/** JSDocs type definitions */

/**
 * Props data to use when rendering a gift options component.
 *
 * @typedef {Object} CmsDynamicBlockTalonProps
 *
 * @property {Boolean} loading Query loading indicator.
 * @property {Object} error Error of the GraphQl query.
 * @property {Object} data Data returned by the query.
 */

/**
 * This is a type used by the {@link useCmsDynamicBlock} talon.
 *
 * @typedef {Object} CmsDynamicBlockOperations
 *
 * @property {GraphQLAST} getSalesRulesDataQuery get sales rules data from cart.
 * @property {GraphQLAST} getCmsDynamicBlocksQuery get cms dynamics blocks.
 */
