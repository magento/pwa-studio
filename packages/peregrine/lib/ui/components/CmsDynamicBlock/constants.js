export const DISPLAY_MODE_FIXED_TYPE = `fixed`;
export const DYNAMIC_BLOCK_FIXED_TYPE = `SPECIFIED`;
export const DISPLAY_MODE_SALES_RULE_TYPE = `salesrule`;
export const DYNAMIC_BLOCK_SALES_RULE_TYPE = `CART_PRICE_RULE_RELATED`;
export const DISPLAY_MODE_CATALOG_RULE_TYPE = `catalogrule`;
export const DYNAMIC_BLOCK_CATALOG_RULE_TYPE = `CATALOG_PRICE_RULE_RELATED`;
