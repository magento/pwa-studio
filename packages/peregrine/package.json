{
  "name": "@magento/peregrine",
  "version": "12.1.0",
  "publishConfig": {
    "access": "public"
  },
  "description": "The core runtime of Magento PWA",
  "main": "lib/index.js",
  "scripts": {
    "clean": " "
  },
  "repository": "github:magento/pwa-studio",
  "author": "Magento Commerce",
  "license": "(OSL-3.0 OR AFL-3.0)",
  "bugs": {
    "url": "https://github.com/magento/pwa-studio/issues"
  },
  "homepage": "https://github.com/magento/pwa-studio/tree/master/packages/peregrine#readme",
  "dependencies": {
    "fast-glob": "~3.2.4"
  },
  "devDependencies": {
    "@apollo/client": "~3.4.0",
    "@magento/eslint-config": "~1.5.0",
    "@testing-library/react-hooks": "~5.0.3",
    "intl": "~1.2.5",
    "intl-locales-supported": "~1.8.12",
    "react": "~17.0.1",
    "react-dom": "~17.0.1",
    "react-intl": "~5.20.0",
    "react-redux": "~7.2.2",
    "react-router-dom": "~5.2.0",
    "react-test-renderer": "~17.0.1",
    "redux": "~4.0.4",
    "redux-actions": "~2.6.5",
    "wait-for-expect": "~1.2.0"
  },
  "peerDependencies": {
    "@apollo/client": "~3.4.0",
    "@babel/runtime": "~7.15.3",
    "informed": "~3.29.0",
    "react": "~17.0.1",
    "react-dom": "~17.0.1",
    "react-intl": "~5.20.0",
    "react-redux": "~7.2.2",
    "react-router-dom": "~5.2.0",
    "redux": "~4.0.4",
    "redux-actions": "~2.6.5"
  },
  "engines": {
    "node": ">=14.x",
    "yarn": ">=1.12.0"
  },
  "pwa-studio": {
    "targets": {
      "declare": "./lib/targets/peregrine-declare",
      "intercept": "./lib/targets/peregrine-intercept"
    }
  },
  "module": "lib/index.js",
  "jsnext:main": "lib/index.js",
  "es2015": "lib/index.js",
  "sideEffects": false
}
