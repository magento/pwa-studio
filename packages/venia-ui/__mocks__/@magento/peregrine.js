import React from 'react';
import { BrowserPersistence } from './util/simplePersistence';

const peregrine = jest.requireActual('../../../peregrine/lib');

// re-exports
const RestApi = {
    Magento2: {
        request: jest.fn()
    }
};

const Util = { BrowserPersistence };

// hooks
const useEventListener = jest.fn(peregrine.useEventListener);
const useDropdown = jest.fn(peregrine.useDropdown);
const useSearchParam = jest.fn(peregrine.useSearchParam);
const useWindowSize = jest.fn(peregrine.useWindowSize);
const useToasts = jest.fn(peregrine.useToasts);
const useTypePolicies = jest.fn(peregrine.useTypePolicies);
// components

/**
 * the Price component from @magento/peregrine
 * has browser-specific functionality and cannot
 * currently by rendered in the test environment
 */
const Price = jest.fn().mockReturnValue(<div />);

module.exports = {
    ...peregrine,
    Price,
    RestApi,
    Util,
    useEventListener,
    useDropdown,
    useSearchParam,
    useWindowSize,
    useToasts,
    useTypePolicies
};
