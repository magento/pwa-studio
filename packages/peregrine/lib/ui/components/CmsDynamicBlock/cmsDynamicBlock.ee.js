import React from 'react';
import { array, oneOf, oneOfType, string } from 'prop-types';

import { useCmsDynamicBlock } from '@magento/peregrine/lib/talons/CmsDynamicBlock/useCmsDynamicBlock';
import ErrorView from '@magento/venia-ui/lib/components/ErrorView';

import {
    DISPLAY_MODE_FIXED_TYPE,
    DYNAMIC_BLOCK_FIXED_TYPE,
    DISPLAY_MODE_SALES_RULE_TYPE,
    DYNAMIC_BLOCK_SALES_RULE_TYPE,
    DISPLAY_MODE_CATALOG_RULE_TYPE,
    DYNAMIC_BLOCK_CATALOG_RULE_TYPE
} from './constants';
import DynamicBlock from './dynamicBlock';

const getDynamicBlockType = displayMode => {
    if (displayMode === DISPLAY_MODE_FIXED_TYPE) {
        return DYNAMIC_BLOCK_FIXED_TYPE;
    } else if (displayMode === DISPLAY_MODE_SALES_RULE_TYPE) {
        return DYNAMIC_BLOCK_SALES_RULE_TYPE;
    } else if (displayMode === DISPLAY_MODE_CATALOG_RULE_TYPE) {
        return DYNAMIC_BLOCK_CATALOG_RULE_TYPE;
    }

    return DYNAMIC_BLOCK_FIXED_TYPE;
};

const CmsDynamicBlockGroup = props => {
    const { displayMode } = props;
    const type = getDynamicBlockType(displayMode);

    const { loading, error, data } = useCmsDynamicBlock({
        ...props,
        type
    });

    if (!data) {
        if (loading) {
            return null;
        }

        if (error) {
            return <ErrorView message={error.message} />;
        }
    }

    const { items } = data?.dynamicBlocks || {};

    if (!Array.isArray(items) || !items.length) {
        return null;
    }

    return items.map((item, index) => (
        <DynamicBlock key={item.uid} index={index} {...item} />
    ));
};

CmsDynamicBlockGroup.defaultProps = {
    displayMode: 'fixed'
};

CmsDynamicBlockGroup.propTypes = {
    displayMode: oneOf([
        DISPLAY_MODE_FIXED_TYPE,
        DISPLAY_MODE_SALES_RULE_TYPE,
        DISPLAY_MODE_CATALOG_RULE_TYPE
    ]),
    locations: array,
    uids: oneOfType([string, array]).isRequired
};

export default CmsDynamicBlockGroup;
