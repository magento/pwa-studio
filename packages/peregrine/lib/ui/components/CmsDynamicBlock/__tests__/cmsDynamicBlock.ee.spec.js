import React from 'react';

import { createTestInstance } from '@magento/peregrine';
import { useCmsDynamicBlock } from '@magento/peregrine/lib/talons/CmsDynamicBlock/useCmsDynamicBlock';

import {
    DISPLAY_MODE_FIXED_TYPE,
    DISPLAY_MODE_SALES_RULE_TYPE,
    DISPLAY_MODE_CATALOG_RULE_TYPE
} from '../constants';
import CmsDynamicBlock from '../cmsDynamicBlock.ee';

const mockUids = 'uids';

jest.mock('@magento/peregrine/lib/talons/CmsDynamicBlock/useCmsDynamicBlock');
jest.mock('@magento/venia-ui/lib/components/ErrorView', () => props => (
    <mock-ErrorView {...props} />
));
jest.mock('../dynamicBlock', () => props => <mock-DynamicBlock {...props} />);

let inputProps = {};

const Component = () => {
    return <CmsDynamicBlock {...inputProps} />;
};

const givenDefaultValues = () => {
    inputProps = {
        displayMode: DISPLAY_MODE_FIXED_TYPE,
        uids: mockUids
    };
};

const givenCatalogRuleType = () => {
    inputProps = {
        displayMode: DISPLAY_MODE_CATALOG_RULE_TYPE,
        uids: mockUids
    };
};

const givenSalesRuleType = () => {
    inputProps = {
        displayMode: DISPLAY_MODE_SALES_RULE_TYPE,
        uids: mockUids
    };
};

const givenUnknownType = () => {
    inputProps = {
        displayMode: 'test',
        uids: mockUids
    };
};

describe('#CmsDynamicBlock AC', () => {
    beforeEach(() => {
        givenDefaultValues();
    });

    it('renders null when loading', () => {
        useCmsDynamicBlock.mockReturnValueOnce({
            data: null,
            error: undefined,
            loading: true
        });

        const component = createTestInstance(<Component />);

        expect(component.toJSON()).toMatchSnapshot();
    });

    it('renders ErrorView when error', () => {
        useCmsDynamicBlock.mockReturnValueOnce({
            data: null,
            error: {
                message: 'Error'
            },
            loading: false
        });

        const component = createTestInstance(<Component />);

        expect(component.toJSON()).toMatchSnapshot();
    });

    it('renders null when no data', () => {
        useCmsDynamicBlock.mockReturnValueOnce({
            data: null,
            error: undefined,
            loading: false
        });

        const component = createTestInstance(<Component />);

        expect(component.toJSON()).toMatchSnapshot();
    });

    it('renders null when items are empty', () => {
        useCmsDynamicBlock.mockReturnValue({
            data: {
                dynamicBlocks: {
                    items: []
                }
            },
            error: undefined,
            loading: false
        });

        const component = createTestInstance(<Component />);

        expect(component.toJSON()).toMatchSnapshot();
    });

    it('renders DynamicBlock when data with type fixed', () => {
        useCmsDynamicBlock.mockReturnValue({
            data: {
                dynamicBlocks: {
                    items: [
                        {
                            uid: 'itemUid1',
                            content: {
                                html: '<div>Item 1</div>'
                            }
                        }
                    ]
                }
            },
            error: undefined,
            loading: false
        });

        const component = createTestInstance(<Component />);

        expect(component.toJSON()).toMatchSnapshot();
    });

    it('renders DynamicBlock when data with type catalogrule', () => {
        givenCatalogRuleType();

        useCmsDynamicBlock.mockReturnValue({
            data: {
                dynamicBlocks: {
                    items: [
                        {
                            uid: 'itemUid1',
                            content: {
                                html: '<div>Item 1</div>'
                            }
                        }
                    ]
                }
            },
            error: undefined,
            loading: false
        });

        const component = createTestInstance(<Component />);

        expect(component.toJSON()).toMatchSnapshot();
    });

    it('renders DynamicBlock when data with type salesrule', () => {
        givenSalesRuleType();

        useCmsDynamicBlock.mockReturnValue({
            data: {
                dynamicBlocks: {
                    items: [
                        {
                            uid: 'itemUid1',
                            content: {
                                html: '<div>Item 1</div>'
                            }
                        }
                    ]
                }
            },
            error: undefined,
            loading: false
        });

        const component = createTestInstance(<Component />);

        expect(component.toJSON()).toMatchSnapshot();
    });

    it('renders DynamicBlock when data with unknown type', () => {
        givenUnknownType();

        useCmsDynamicBlock.mockReturnValue({
            data: {
                dynamicBlocks: {
                    items: [
                        {
                            uid: 'itemUid1',
                            content: {
                                html: '<div>Item 1</div>'
                            }
                        }
                    ]
                }
            },
            error: undefined,
            loading: false
        });

        const component = createTestInstance(<Component />);

        expect(component.toJSON()).toMatchSnapshot();
    });
});
