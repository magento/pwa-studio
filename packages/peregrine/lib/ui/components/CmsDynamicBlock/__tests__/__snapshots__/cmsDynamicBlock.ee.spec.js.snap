// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#CmsDynamicBlock AC renders DynamicBlock when data with type catalogrule 1`] = `
<mock-DynamicBlock
  content={
    Object {
      "html": "<div>Item 1</div>",
    }
  }
  index={0}
  uid="itemUid1"
/>
`;

exports[`#CmsDynamicBlock AC renders DynamicBlock when data with type fixed 1`] = `
<mock-DynamicBlock
  content={
    Object {
      "html": "<div>Item 1</div>",
    }
  }
  index={0}
  uid="itemUid1"
/>
`;

exports[`#CmsDynamicBlock AC renders DynamicBlock when data with type salesrule 1`] = `
<mock-DynamicBlock
  content={
    Object {
      "html": "<div>Item 1</div>",
    }
  }
  index={0}
  uid="itemUid1"
/>
`;

exports[`#CmsDynamicBlock AC renders DynamicBlock when data with unknown type 1`] = `
<mock-DynamicBlock
  content={
    Object {
      "html": "<div>Item 1</div>",
    }
  }
  index={0}
  uid="itemUid1"
/>
`;

exports[`#CmsDynamicBlock AC renders ErrorView when error 1`] = `
<mock-ErrorView
  message="Error"
/>
`;

exports[`#CmsDynamicBlock AC renders null when items are empty 1`] = `null`;

exports[`#CmsDynamicBlock AC renders null when loading 1`] = `null`;

exports[`#CmsDynamicBlock AC renders null when no data 1`] = `null`;
